kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -980,4 +973,30 @@ private void printLogForRetryableWithConflictException(
         commitAsVersion,
         cfe);
   }
+
+  private void createDeltaLogAndStagedCommitsDirectoriesIfNeeded(
```

Comment:
What will happen if we have a catalog table without the _staged_commits directory?

Should we check for that and either (1) create the directory anyways or (2) throw invalid table exception?

Reply:
Kernel already has the `engine` and can make the directory. No need to put this on the Committer.